identity-apps-core/apps/authentication-portal/src/main/webapp/domain.jsp

```diff
@@ -28,14 +28,17 @@
 
 <%
     String domainUnknown = AuthenticationEndpointUtil.i18n(resourceBundle, "domain.unknown");
-    String errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "authentication.failed");
+    String errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "authentication.failed.please.retry");
     boolean loginFailed = false;
     if (Boolean.parseBoolean(request.getParameter("authFailure"))) {
         loginFailed = true;
         if (request.getParameter("authFailureMsg") != null) {
-            errorMessage = request.getParameter("authFailureMsg");
-
-            if (domainUnknown.equalsIgnoreCase(errorMessage)) {
+            String error = request.getParameter("authFailureMsg");
+            if (!error.equalsIgnoreCase(AuthenticationEndpointUtil.i18n(resourceBundle, error))) {
+                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, error);
+            }
```

Comment:
Could you kindly explain what we are doing here?

Reply:
We will only display the error message from the request if it matches with a key defined in the resourceBundle [1]. The same validation is performed in some other pages as well [2] to prevent potential phishing attacks. 

[1] - https://github.com/wso2/identity-apps/blob/%40wso2is/identity-apps-core%402.4.2/identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties

[2] - https://github.com/wso2/identity-apps/blob/%40wso2is/identity-apps-core%402.4.2/identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp#L161